### PR TITLE
Open Account API

### PR DESCRIPTION
**Pull request by wood-archer**
Implements API to open an account. 

API Flow
---
```
Router               # POST /api/accounts
|> Controller        # create(conn, %{"account" => %{"initial_balance": initial_balance})
|> Context           # Accounts.open_account(%{"initial_balance" => initial_balance})
|> Command           # %OpenAccount{account_uuid: uuid, initial_balance: initial_balance}         
|> Dispatcher        # dispatch(%OpenAccount{}, to: Account) 
|> Aggregate         # Account.execute(Account, %OpenAccount{})
|> Event             # %AccountOpened{account_uuid: uuid, initial_balance: initial_balance}     
|> Projector         # project(%AccountOpened{} = evt, fn multi -> Ecto.Multi.insert(..) end)


* OpenAccount        - Command 
* Account            - Aggregate
* AccountOpened      - Event
```

Diagram
---
image

### CODE REVIEW

File: lib/bank_api/application.ex [hunk 1]
```diff
@@ -14,7 +14,9 @@ defmodule BankAPI.Application do
       # Start the PubSub system
       {Phoenix.PubSub, name: BankAPI.PubSub},
       # Start the Endpoint (http/https)
-      BankAPIWeb.Endpoint
+      BankAPIWeb.Endpoint,
+      # Start EventStore
+      BankAPI.EventStore
```
Comment (vasspilka) [inline lib/bank_api/application.ex:19]: You also have to start the commanded application

Comment (vasspilka) [reply lib/bank_api/application.ex:19]: BankAPI.Application

Comment (wood-archer) [reply lib/bank_api/application.ex:19]: You mean to add this on top of the application module?
```
use Commanded.Application, otp_app: :bank_api
```

Comment (wood-archer) [reply lib/bank_api/application.ex:19]: Oh, I think I know what you mean. We need a separate module to start the commanded application and then we add it to the supervision tree.

**Comment by wood-archer** [comment]
@vasspilka I started to implement the api to open an account but got stuck in this error:
```
** (RuntimeError) could not lookup nil because it was not started or it does not exist
```
Couldn't make the tests pass yet.

**Pull request merged**